llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -15931,6 +15931,26 @@ static OptimizationRemark emitAtomicRMWLegalRemark(const AtomicRMWInst *RMW) {
          << " operation at memory scope " << MemScope;
 }
 
+static bool isHalf2OrBFloat2(Type *Ty) {
```

Comment:
Does the underlying type really matter? Is 2 x 16-bit type sufficient?

Reply:
Both instructions were added together. The currently defined feature is HasAtomicFlatPkAdd16Insts, so this mirrors that